### PR TITLE
Vendor `jquery.are-you-sure` with strict mode fixes

### PR DESCRIPTION
**Pull request by silverwind**
Extract from https://github.com/go-gitea/gitea/pull/25940 and because https://github.com/go-gitea/gitea/pull/26743 does seem to need more work.

This will be required if we are to run our JS in [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode).

Previously, the two variables `$fields` and `$dirtyForms` polluted `window`:

image

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: web_src/js/features/common-global.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import $ from 'jquery';
-import 'jquery.are-you-sure';
+import '../vendor/ays.js';
```
Comment (wxiaoguang) [inline web_src/js/features/common-global.js]: I would prefer to use `../vendor/jquery.are-you-sure.js` to match its package name.

Comment (silverwind) [reply web_src/js/features/common-global.js]: Done

File: web_src/js/vendor/jquery.are-you-sure.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
+// Fork of the upstream module. The only changes are the addtion of `const` on
```
Comment (silverwind) [inline web_src/js/vendor/jquery.are-you-sure.js]:
```suggestion
// Fork of the upstream module. The only changes are the addition of `const` on
```

File: web_src/js/vendor/jquery.are-you-sure.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
+// Fork of the upstream module. The only changes are the addition of `const` on
+// lines 93 and 161 to make it strict-mode compatible.
```
Comment (silverwind) [inline web_src/js/vendor/jquery.are-you-sure.js]:
```suggestion
// lines 93 and 161 to make it strict mode compatible.
```

**Review by delvh — APPROVED** [review]

**Pull request merged**